src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int SIZE_INCREASE_FACTOR = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        growIfHashMapFull();
+        int index = getValidIndex(key, table.length);
+        Node<K, V> currentNode = table[index];
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (currentNode == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> nodeWithUniqueKey = updateIfKeysEqual(currentNode, key, value);
+            if (nodeWithUniqueKey != null) {
+                nodeWithUniqueKey.next = newNode;
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getValidIndex(key, table.length);
+        Node<K, V> node = getNode(index);
+        if (node != null) {
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void growIfHashMapFull() {
+        if (size > table.length * LOAD_FACTOR) {
+            resize();
+        }
+    }
+
+    private int getValidIndex(K key, int capacity) {
+        return (capacity - 1) & hash(key,capacity);
+    }
+
+    private int hash(K key, int capacity) {
+        return (key == null) ? 0 : key.hashCode() % capacity;
+    }
+
+    private Node<K, V> updateIfKeysEqual(Node<K, V> node, K key, V value) {
```

Comment:
dont return anything, do required logic in this method
```suggestion
    private void updateIfKeysEqual(Node<K, V> node, K key, V value) {
```